cmd/util/ledger/migrations/accounts.go

```diff
@@ -0,0 +1,139 @@
+package migrations
+
+import (
+	"encoding/hex"
+	"strings"
+
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+func AddMissingKeysMigration(payloads []ledger.Payload) ([]ledger.Payload, error) {
+	l := newLed(payloads)
+	a := state.NewAccounts(l)
+
+	coreContractEncodedKey := "f847b8402b0bf247520770a4bad19e07f6d6b1e8f0542da564154087e2681b175b4432ec2c7b09a52d34dabe0a887ea0f96b067e52c6a0792dcff730fe78a6c5fbbf0a9c02038203e8"
+
+	// Testnet FungibleToken
+	err := appendKeyForAccount(a, "9a0766d93b6608b7", coreContractEncodedKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// Testnet NonFungibleToken
+	err = appendKeyForAccount(a, "631e88ae7f1d7c20", coreContractEncodedKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// Testnet FlowToken
+	err = appendKeyForAccount(a, "7e60df042a9c0868", coreContractEncodedKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// Testnet FlowFees
+	err = appendKeyForAccount(a, "912d5440f7e3769e", coreContractEncodedKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// Testnet Staking
+	err = appendKeyForAccount(a, "9eca2b38b18b5dfe", coreContractEncodedKey)
+	if err != nil {
+		return nil, err
+	}
+
+	nonCoreContractEncodedKey := "f847b840a272d78cfa14eb248d95c12da8c6a24db9fda5ceddc07444080b49ef6cd15a06e88223af8acd235e2ff7a627adb81cf37d0a1384d9985de4bc7a7fc7eb86848402038203e8"
+
+	// Testnet StakingProxy
+	err = appendKeyForAccount(a, "7aad92e5a0715d21", nonCoreContractEncodedKey)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Testnet LockedTokens
+	err = appendKeyForAccount(a, "95e019a17d0e23d7", nonCoreContractEncodedKey)
+
+	if err != nil {
+		return nil, err
+	}
+	return l.Payloads(), nil
+}
+
+func appendKeyForAccount(accounts *state.Accounts, addressInHex string, encodedKeyInHex string) error {
+	// fungible token
+	add := flow.HexToAddress(addressInHex)
+	ok, err := accounts.Exists(add)
+	if err != nil {
+		return err
+	}
+	// Core Contracts
+	if ok {
+		accountKeyBytes, err := hex.DecodeString(encodedKeyInHex)
+		if err != nil {
+			return err
+		}
+		accountKey, err := flow.DecodeRuntimeAccountPublicKey(accountKeyBytes, 0)
+		if err != nil {
+			return err
+		}
+		err = accounts.AppendPublicKey(add, accountKey)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type led struct {
```

Comment:
Just to make sure I understand, what's the purpose of this `led` struct?

Reply:
Is this something that will be used by all migrations? It looks like it simply captures all writes and constructs a payload for each?

Reply:
it is only intended for this migration, don't think is valuable to keep it long term.

Reply:
ledger migration is basically a method of []payload to []payload. this led is basically an abstraction on top of ledger migration to perform account operations, I recommend we remove it after this migration to prevent these operations to become a common practice.